src/components/Combobox/Combobox.jsx

```diff
@@ -175,6 +141,42 @@ const Combobox = forwardRef(
       [activeCategoryLabel]
     );
 
+    const { items, itemsMap, selectableItems } = useItemsData({
+      categories,
+      options: filteredOptions,
+      filterValue,
+      withCategoriesDivider,
+      optionLineHeight
+    });
+
+    const overrideOnClick = useCallback(
```

Comment:
this is being used only be the `useKeyboardNavigation` why not move this logic inside ?

Reply:
I send explanation on slack, let me know if there is an issue and I will merge in a different pr :)